### ISSUE TITLE
Remix fails to run with netlify-cli@14.2.1 - TypeError: Cannot read properties of undefined (reading 'v2_errorBoundary')

### ISSUE THREAD

**Issue by tcschiller** [remix-run/remix#6137]
### What version of Remix are you using?

1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. Upgrade `netlify-cli` to latest (`14.2.1` at the time of this writing).
2. Attempt to run Remix with whatever `package.json` command in your project invokes `netlify dev`.
3. Observe CLI error.
4. Load in browser and observe `The function has crashed` error.

### Expected Behavior

Project is able to run.

### Actual Behavior

Stack trace:

`TypeError: Cannot read properties of undefined (reading 'v2_errorBoundary')`

```
~/my-project/node_modules/@remix-run/server-runtime/dist/routes.js:42:35
Array.map ()
Object.createStaticHandlerDataRoutes (~/Projects/my-project/node_modules/@remix-run/server-runtime/dist/routes.js:41:45)
Object.createRequestHandler (~/Projects/my-project/node_modules/@remix-run/server-runtime/dist/server.js:28:27)
createRequestHandler (~/Projects/my-project/node_modules/@remix-run/netlify/dist/server.js:31:28)
Object. (~/Projects/my-project/server.js:43:24)
Module._compile (node:internal/modules/cjs/loader:1226:14)
Object.Module._extensions..js (node:internal/modules/cjs/loader:1280:10)
Module.load (node:internal/modules/cjs/loader:1089:32)
Function.Module._load (node:internal/modules/cjs/loader:930:12)
```

**Comment by machour** [comment]
This is most likely due to one of your remix dependencies not being at the same versions of the others.
Can you provide your package.json dependencies & devDependencies ?

**Comment by tcschiller** [comment]
`npm list` results of my working project:
```
├── @netlify/functions@1.4.0
├── @remix-run/dev@1.15.0
├── @remix-run/netlify@1.15.0
├── @remix-run/node@1.15.0
├── @remix-run/react@1.15.0
├── @types/react-dom@18.0.11
├── @types/react@18.0.38
├── concurrently@8.0.1
├── cross-env@7.0.3
├── eslint@8.39.0
├── husky@8.0.3
├── jest@29.5.0
├── netlify-cli@13.2.2
├── react-dom@18.2.0
├── react@18.2.0
├── tailwindcss@3.3.1
└── typescript@5.0.4
```
As far as I can tell, we should be good from a Remix package version match standpoint.

**Comment by tcschiller** [comment]
Interesting, I can't reproduce this with the `npx create-remix@latest` command.

**Comment by machour** [comment]
Could you try explicitly listing @remix-run/server-runtime in your dependencies? (At the same version ofc)

**Comment by tcschiller** [comment]
> Could you try explicitly listing @remix-run/server-runtime in your dependencies? (At the same version ofc)

The same error occurs after that change.

**Comment by tcschiller** [comment]
I'm happy to try a few things on my end. I suspect it's an artifact of having an old Remix app upgraded many, many times in these early days! Something is still stuck in the past.

**Comment by tcschiller** [comment]
Yup, I cleaned up an old `netlify` directory and updated some configs, and things run now on latest `netlify-cli`. Thanks, closing this out!

**Comment by nunoci** [comment]
@juliaschiller150 you mind to share what you fixed, im having the same issue on remix 1.7.9 with @remix-run/netifly@1.7.9

`typeError: Cannot read properties of undefined (reading 'v2_errorBoundary')
    at /var/task/node_modules/@remix-run/netlify/node_modules/@remix-run/server-runtime/dist/routes.js:46:35
    at Array.map (<anonymous>)
    at Object.createStaticHandlerDataRoutes (/var/task/node_modules/@remix-run/netlify/node_modules/@remix-run/server-runtime/dist/routes.js:45:45)
    at Object.createRequestHandler (/var/task/node_modules/@remix-run/netlify/node_modules/@remix-run/server-runtime/dist/server.js:28:27)
    at createRequestHandler (/var/task/node_modules/@remix-run/netlify/dist/server.js:31:28)
    at Object.<anonymous> (/var/task/.netlify/functions-internal/server.js:11231:55)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1013:12)`

**Comment by juliaschiller150** [comment]
I'm afraid I can't remember the specifics beyond the above, @nunoci. I would recommend spinning up a fresh Remix install in a sibling directory and diffing netlify related directories and Remix configs between that new install and your existing app.

Let us know what you find!
